x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -172,6 +174,14 @@ public synchronized AutoFollowStats getStats() {
     }
 
     synchronized void updateStats(List<AutoFollowResult> results) {
+        // purge stats for removed patterns
+        var currentPatterns = this.patterns;
+        recentAutoFollowErrors.keySet().removeIf(key -> {
+            var semicolonPosition = key.indexOf(':');
```

Comment:
I think we should use `lastIndexOf`, since index-names cannot have `:`, but unfortunately, I think autofollow pattern names can (at least I did not find the protection against it, hope you will test it).